js/ai/src/generate.ts

```diff
@@ -376,10 +380,14 @@ export async function generate<
     registry,
     stripNoop(resolvedOptions.onChunk ?? resolvedOptions.streamingCallback),
     async () => {
-      const response = await generateHelper(registry, {
-        rawRequest: params,
-        middleware: resolvedOptions.use,
-      });
+      const generateFn = () =>
+        generateHelper(registry, {
+          rawRequest: params,
+          middleware: resolvedOptions.use,
+        });
+      const response = await (resolvedOptions.context
+        ? runWithContext(registry, resolvedOptions.context, generateFn)
+        : generateFn());
```

Comment:
Would it be better to have `runWithContext` handle an `undefined` context so that this doesn't need to be a ternary?

Reply:
Not sure... I'm thinking we might want the ability to set undefined context (run some action in isolation)...

Another option is:
```ts
      const response = await runWithContext(registry, resolvedOptions.context || getContext(registry), generateFn); 
```

but it would mean we're unnecessarily setting context in ALS, avoidable overhead.

Reply:
I guess I'm saying that `runWithContext` could see an `undefined` context and no-op -- not actually do anything with ALS and just fall through to calling the underlying function. So same behavior as defined here, but encapsulated in a single place.

```ts
runWithContext(registry, undefined, generateFn) // equivalent to generateFn()
runWithContext(registry, {}, generateFn) // stores empty context in ALS and runs geneateFn()
runWithContext(registry, someContext, generateFn) // stores context in ALS and runs generateFn()
```

Reply:
Done.
I'm personally usually wary of these patterns because it turns `runWithContext` into `runWithContextExceptIfYouPassUndefinedContextThenItsAPassthough`, which may not be obvious to the caller. But I agree it makes call sites much cleaner.